runtime/testy/Ansible.py

```diff
@@ -0,0 +1,199 @@
+        self.get_index = 1 - self.get_index
+
+    def get(self):
+        return self.data[self.get_index]
+
+
+class AnsibleHandler():
+    """Parent class for UDP Processes that spawns threads 
+
+    Initializes generalized instance variables for both UDP sender and receiver, and creates a callable
+    method to initialize the two threads per UDP process and start them. 
+    """
+    def __init__(self, packagerName, packagerThread, socketName, socketThread,
+                 badThingsQueue, stateQueue, pipe):
+        self.packagerFunc = packagerThread
+        self.socketFunc = socketThread
+        self.badThingsQueue = badThingsQueue
+        self.stateQueue = stateQueue
+        self.pipe = pipe
+        self.packagerName = packagerName
+        self.socketName = socketName
+
+    def threadMaker(self, threadTarget, threadName):
+        thread = threading.Thread(target=threadTarget,
+                                  name=threadName,
+                                  args=(self, self.badThingsQueue, self.stateQueue, self.pipe))
+        thread.daemon = True
+        return thread
+
+    def start(self):
+        packagerThread = self.threadMaker(self.packagerFunc, self.packagerName)
+        socketThread = self.threadMaker(self.socketFunc, self.socketName)
+        packagerThread.start()
+        socketThread.start()
+        packagerThread.join()
+        socketThread.join()
+
+class UDPSendClass(AnsibleHandler):
+    SEND_PORT = 1235
+    packagerHZ = 20.0
+    socketHZ = 20.0
+
+    def __init__(self, badThingsQueue, stateQueue, pipe):
+        self.sendBuffer = TwoBuffer()
+        packagerName = THREAD_NAMES.UDP_PACKAGER
+        sockSendName = THREAD_NAMES.UDP_SENDER
+        super().__init__(packagerName, UDPSendClass.packageData, sockSendName,
+                         UDPSendClass.udpSender, badThingsQueue, stateQueue, pipe)
+
+    def packageData(self, badThingsQueue, stateQueue, pipe):
+        """Function run as a thread that packages data to be sent.
+
+        The robot's current state is received from the StateManager via the pipe and packaged 
+        by the package function, defined internally. The packaged data is then placed 
+        back into the TwoBuffer replacing the previous state.
+        """
+        def package(state):
+            """Helper function that packages the current state.
+
+            Currently this function converts the input into bytes. This will
+            eventually be implemented to package the rawState into protos.
+            """
+            s = "TEST" 
+            b = bytearray(map(ord, s))
+            return b 
+
+        while True:
+            try:
+                nextCall = time.time()
+                stateQueue.put([SM_COMMANDS.SEND_ANSIBLE, []])
+                rawState = pipe.recv()
+                packState = package(rawState)
+                self.sendBuffer.replace(packState)
+                nextCall += 1.0/self.packagerHZ
```

Comment:
Doess the initialization of nextCall each loop not catch this as it only times how long each iteration of the loop takes? If it does take too long, it would never sleep but that would be the correct behavior as we only want to limit the maximum rate it goes right?